pamelalozano/src/main/webapp/script.js

```diff
@@ -27,14 +27,9 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-function projectToggle(id){
-    //the id for each button is proj[i], i being the index
-    //c-proj[i] is the id for the div with the description of each project
-
-    if($("#c-"+id)[0].style.display!="block"){
-        $("#c-"+id)[0].style.display="block"
-        location.href = "#c-"+id;
-    } else{
-          $("#c-"+id)[0].style.display="none"
-    };
-}
\ No newline at end of file
+document.querySelectorAll('.project').forEach(item => {
```

Comment:
This is definitely an improvement over the last version!

There is still room for improvement by using event delegation and bubbling i.e. put a single event listener on the projects parent container element (div with ID "#projects") and let the click events on its children bubble up to its event listener. In that event handler callback you'll have to figure out which child element was actually clicked by looking at `event.target` and possibly some other properties of event.target to figure out if you need to do anything.

Very good examples here: https://davidwalsh.name/event-delegate

Reply:
Solved!